### DISCUSSION TITLE
17.1 has been released

### DISCUSSION THREAD

**Discussion by bbakerman** [graphql-java/graphql-java#2509] [General]
https://github.com/graphql-java/graphql-java/releases/tag/v17.1

This release upgrade the DataLoader library to 3.1.0 which adds the ability to have an external value cache in place during data loader batch calls.

You can use it to model access to external caches like REDIS amd even do batch "cache gets".

Originally planned for 17.0 the DataLoader update has to be reverted due to a late design problem we found.

This has been fixed and is now ready for use

**Comment by sp00m** [comment]
Hey, what has been packaged in dataloader's v3.1.0 exactly? I can indeed find it in [Maven Central](https://search.maven.org/artifact/com.graphql-java/java-dataloader/3.1.0/jar), but no associated [release](https://github.com/graphql-java/java-dataloader/releases) nor [tag](https://github.com/graphql-java/java-dataloader/tags).

**Reply by bbakerman** [reply]
Sorry I staged up a  draft release but forgot to hit publish once the artifact hit maven

https://github.com/graphql-java/java-dataloader/releases/tag/3.1.0
